src/main/webapp/app/overview/overview.component.ts

```diff
@@ -1,28 +1,50 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnDestroy } from '@angular/core';
 import { Course, CourseScoreCalculationService, CourseService } from 'app/entities/course';
 import { HttpResponse } from '@angular/common/http';
 import { JhiAlertService } from 'ng-jhipster';
 import { Exercise, ExerciseService } from 'app/entities/exercise';
 import { AccountService } from 'app/core';
-import { TUM_REGEX } from 'app/app.constants';
+import { GuidedTour } from 'app/guided-tour/guided-tour.constants';
+import { GuidedTourService } from 'app/guided-tour/guided-tour.service';
+import { Subscription } from 'rxjs';
 
 @Component({
     selector: 'jhi-overview',
     templateUrl: './overview.component.html',
-    styles: [],
+    styleUrls: [],
 })
-export class OverviewComponent {
+export class OverviewComponent implements OnDestroy {
     public courses: Course[];
     public nextRelevantCourse: Course;
+    public overviewTour: GuidedTour;
+
+    subscription: Subscription;
 
     constructor(
         private courseService: CourseService,
         private exerciseService: ExerciseService,
         private jhiAlertService: JhiAlertService,
         private accountService: AccountService,
         private courseScoreCalculationService: CourseScoreCalculationService,
+        private guidedTourService: GuidedTourService,
     ) {
         this.loadAndFilterCourses();
+
+        setTimeout(() => {
```

Comment:
Also I think we should rather do this in the ngOnInit / ngOnChanges and not in the constructor.